src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -724,12 +725,22 @@ protected override bool EnsureIncludeSerializer()
         protected override bool EnsureIncludeDeserializer()
         {
             // TODO -- this should always return true when use model reader writer is enabled.
-            return Configuration.UseModelReaderWriter || _usage.HasFlag(InputModelTypeUsage.Output);
+            return !MgmtReferenceType.IsReferenceType(InputModel) && (Configuration.UseModelReaderWriter || _usage.HasFlag(InputModelTypeUsage.Output));
         }
 
         protected override JsonObjectSerialization? BuildJsonSerialization()
         {
-            return MgmtReferenceType.IsReferenceType(InputModel) || !InputModel.Serialization.Json ? null : _serializationBuilder.BuildJsonObjectSerialization(InputModel, this);
+            return !InputModel.Serialization.Json ? null : _serializationBuilder.BuildJsonObjectSerialization(InputModel, this);
+        }
+
+        protected override ObjectTypeSerialization BuildSerialization()
+        {
+            bool isReference = MgmtReferenceType.IsReferenceType(InputModel);
+            var json = BuildJsonSerialization();
+            var xml = isReference ? null : BuildXmlSerialization();
+            var bicep = isReference ? null : BuildBicepSerialization(json);
+            var multipart = isReference ? null : BuildMultipartSerialization();
+            return new ObjectTypeSerialization(this, json, xml, bicep, multipart, isReference);
```

Comment:
how about we input interfaces instead like others? So that, isReference is kept within SchemaObjectType.